### DISCUSSION TITLE
평평한 곳에서 카메라를 회전하면 카메라가 땅을 뚫어 버립니다.

### DISCUSSION THREAD

**Discussion by peterkangdev** [naverz/zepeto-studio-kor#1256] [5. Q&A]
World: 1.6.0

경사진 곳을 내려가다보면 카메라가 땅을 뚫어 버립니다.
앵글 변경도 해보고, 이것 저것 카메라도 만져 봐도 
카메라가 땅에 닿는 순간 땅 밑이 보였다가 위로 올라옵니다. 
월드를 만들다가 보니 너무 보기싫습니다.
방법이 있으면 알려주세요.

**Comment by chlguswls32** [comment]
안녕하세요. peterkangdev님.

월드의 경사진곳에해당하는 부분에 유니티 Component중 충돌을 체크해주는 Collider라는 컴포넌트를 넣어주세요.
mesh, box등 여러 종류의 Collider가있습니다. 가장 간편한게는 mesh collider를 사용하면좋고, 형태가 단순하면 boxcollder(성능적으로 더 가볍습니다.)를 사용하면 좋습니다. 

그리고 해당 collider의 layer를 collider로 변경해주시면 충돌이 잘될것같습니다.

**Comment by peterkangdev** [comment]
제가 collider 넣었다는 말씀 안드렸군요.
당연히 넣었구요.
카메라를 위에서 아래로 회전할때 땅(collider)에 닿는 순간 아래를 뚫고 위로 올라옵니다. 
회전 최대치를 기본값으로 두고 (-80 ~ 80)  카메라 줌을 최대한 멀리한 뒤 하늘에서 땅으로 천천히 회전해 보세요. 
땅에 닿으면서 아래로 반쯤 뚫고 (물론 collider 때문에 더이상 아래로 내려가진 않습니다) 
땅 밑이 반쯤 보입니다. 
땅 아래를 전부 막아 놔도 문제이고, 마치 만들다 만 월드 같습니다. 
카메라에 콜라이더를 넣어봤는데.... 더 심각해 지더라구요.
해결 방법이 없을까요? 이대로 출시해야 하나요?

**Reply by YSHFM** [reply]
layer설정은 정상적으로 하셨나욤?

**Comment by peterkangdev** [comment]
다른 분들은 문제가 없는지 모르겠는데.. 일단 설정사항을 보여드리겠습니다.

아래는 제페토 카메라 설정입니다.

<img width="576" alt="스크린샷 2022-09-06 오후 12 06 22" src="https://user-images.githubusercontent.com/30133892/188540479-daf5e4e5-b864-49cb-93ce-1b1197badd04.png">

아래는 카메라 본체 설정입니다.

<img width="588" alt="스크린샷 2022-09-06 오후 12 10 59" src="https://user-images.githubusercontent.com/30133892/188540549-5c1d7cef-c185-4a33-97a2-39123c4e487d.png">


아래는 실제 동영상입니다.

https://user-images.githubusercontent.com/30133892/188541859-ecb74f9f-89e7-4fa9-b022-09dbafaada1f.mp4

**Reply by YSHFM** [reply]
카메라하고 플레이어가 꽤 멀리에서 해서 그런것 같은데 가까이에서도 동일한가여?

fov시야각과 카메라의 look방향과 지면의 각도 차이로 인해 발생된 것처럼 보입니다.
이러한 경우라면, 별도로 카메라의 행렬값을 조정하거나 하는 예외처리 과정이 필요할 수도 있을 것 같습니다.

**Reply by peterkangdev** [reply]
가까이에서도 동일합니다. 
캐릭터 전신이 보일 정도로 가까이 두고 하시면 동일한 현상이 보입니다.
최초 기본값으로 plane 하나 깔고, 캐릭터 불러서 사용해 보시면 100% 재현됩니다.
다른 분들은 잘 사용하고 계시는지요? 
무시하고 사용하시는지 아니면 조정을 하시는지 궁금합니다. 
평평한 곳에서 사용하면 100% 재현되던데... 어떻게들 하시는지 궁금하네요?

**Reply by YSHFM** [reply]
흠...아주 순간적으로 보이긴 합니다만 전 이정도는 무시하는 편입니다.
